examples.sh

```diff
@@ -10,10 +10,23 @@ function line {
     echo -e "\n"
 }
 
+function wait_for_services {
+    echo "Waiting for services..."
+    while true; do
+        STATUS_CODE=$(curl -o /dev/null --silent --head --write-out '%{http_code}\n' 'http://localhost:8085/articles')
+        if [ "$STATUS_CODE" = "200" ]
+        then
+            break
+        fi
```

Comment:
there should be a timeout because this could sleep indefinitely

Reply:
`timeout` accepts an inline script, but it is error-prone to use it; if the function can be made an external script the filename itself can be passed https://github.com/elifesciences/builder-base-formula/blob/89a9de925e66593b378f6e03b701e89235fa0850/elife/docker-scripts/docker-wait-healthy#L14-L24

Reply:
I tried adding it using `timeout` but it did not seem to like using the inline script method from within a `function`. You could perhaps just refactor the `start` `function` to be split and insert the `timeout` stuff in between, but I swapped out the `while` for a `for` with a limit. Will this suffice for the time being?